### ISSUE TITLE
Naming conventions in arrow functions and variables

### ISSUE THREAD

**Issue by dgdecondux** [typescript-eslint/typescript-eslint#2718]
Hi, I'm trying to define a naming convention for variables and arrow functions, however it only works with variables and not arrow functions.

**Repro**

```JSON
{
  "selector": "variable",
  "format": [ "snake_case" ]
},
{
  "selector": "variable",
  "types": ["function"],
  "format": [ "camelCase" ]
},
```

**Expected Result**
```TS
const full_name = 'name'; ->valid
const camelCase = () => {} ->valid
```

**Actual Result**

```TS
const full_name = 'name'; ->valid
const camelCase = () => {} -> Invalid
```

**Comment by bradzacher** [comment]
Unable to repro this against master. The config you provided is working fine.

![image](https://user-images.githubusercontent.com/7462525/97461836-5cefa800-18fb-11eb-8ea9-5c34bd888de9.png)

**Comment by dgdecondux** [comment]
Hello, I tried again and what happens is that when using the two rules, the conventions for the variables do not validate the format, but when using only one, at that moment it validates correctly.

With two rules.
![image](https://user-images.githubusercontent.com/72207523/97481004-342ed900-1922-11eb-8b97-6f177e2436eb.png)
![image](https://user-images.githubusercontent.com/72207523/97481025-3abd5080-1922-11eb-9ecb-3d33ba0ddd2a.png)

With a rule
![image](https://user-images.githubusercontent.com/72207523/97481112-54f72e80-1922-11eb-9310-8f74512a1e09.png)
![image](https://user-images.githubusercontent.com/72207523/97481127-59234c00-1922-11eb-9bdd-ad87772600b3.png)

Thanks!

**Comment by bradzacher** [comment]
_I cannot repro this against master_ - providing the exact same repro config to me does not help!
You will need to provide a repro repo or a more complete, isolated example in order for me to help you.

**Comment by dgdecondux** [comment]
Hi, I share the repo with the problem, I also use the ESLint plugin from vscode.
The idea is that the variables are defined with snake_case, and the arrow functions with camelCase

Thank you!

https://github.com/dgdecondux/test-conventions

**Comment by dgdecondux** [comment]
Hi, I solved it by setting the "parserOptions" property in .eslint.json and setting the includes in tsconfig.json

Thanks!
